src/qumin/representations/frequencies.py

```diff
@@ -0,0 +1,398 @@
+        - A zero value means that there is a measure, which is zero
+
+    When aggregating accross rows, any empty cell yields a uniform distribution
+    for the whole set of rows, whereas zeros are taken into account. This behaviour
+    can be disabled for some functions by passing skipna=True.
+
+    Examples:
+
+        >>> p = fl.Package('tests/data/TestPackage/test.package.json')
+        >>> Frequencies.initialize(p)
+        >>> print(Frequencies.info().to_markdown())
+        | Table   | Source      |   Records |   Sum(f) |   Mean(f) |
+        |:--------|:------------|----------:|---------:|----------:|
+        | forms   | forms_table |        18 |      459 |   28.6875 |
+        | lexemes | forms_table |         4 |      459 |  114.75   |
+        | cells   | forms_table |         3 |      459 |  153      |
+
+
+    Attributes:
+        p (frictionless.Package): package to analyze
+        real (bool): Whether the frequencies are real or fake.
+        source (Dict[str, str]): source used by default for each table.
+            Contains either a value for the source field of a Paralex frequency table,
+            or the name of the table used to extract the frequency.
+        forms (:class:`pandas:pandas.DataFrame`):
+            Table of frequency values associated to a form_id.
+        lexemes (:class:`pandas:pandas.DataFrame`):
+            Table of frequency values associated to a lexeme_id.
+        cells (:class:`pandas:pandas.DataFrame`):
+            Table of frequency values associated to a cell_id.
+        """
+
+    p = None
+    col_names = ["lexeme", "cell", "form"]
+    source = {"cells": None,
+                      "lexemes": None,
+                      "forms": None}
+
+    @classmethod
+    def initialize(cls, package, source=False, **kwargs):
+        """Constructor for Frequencies. We gather and store frequencies
+        for forms, lexemes and cells. Behaviour is the following:
+            - If `real` is `False`, we use the paradigms table to generate a Uniform distribution.
+            - If not, we try to get a frequency column from the tables: form, lexemes, cell
+            - If any of those is missing, we use the frequencies table.
+
+        Arguments:
+            p (frictionless.Package): package to analyze
+            source (Dict[str, str]): name of the source to use when several are available.
+            **kwargs: keyword arguments for frequency reading methods.
+        """
+
+        cls.p = package
+
+        if source:
+            cls.source.update(source)
+
+        cls._read_form_frequencies(**kwargs)
+        cls._read_other_frequencies("lexemes", **kwargs)
+        cls._read_other_frequencies("cells", **kwargs)
+
+    @classmethod
+    def _read_form_frequencies(cls, real=True):
+        """Recover the available information about form frequencies.
+
+        Arguments:
+            real (bool): Whether to use real frequencies or
```

Comment:
Would this be more transparent as `uniform=False`  ?

~~~
    uniform (bool): Whether to replace real frequencies by empty uniform distributions
~~~

However, my impression is that you never actually add uniform distributions here ?

Could the condition also be in the other direction, eliminating the edge case first:

~~~
if uniform:
    < make fake uniform distributions>
elif "frequency" in paradigms.columns:
    <extract from frequency column>
elif cls.p.has_resource("frequencies"):
    <extract from frequency table>
else:
    <error>
~~~

Reply:
I get you, but then how about `force_uniform` ? So that the semantic makes it clear that you can get "default uniform" or "forced uniform". Otherwise, I don't like the idea that I could pass `uniform=False` and still get a fake uniform distribution.